joblib/_parallel_backends.py

```diff
@@ -170,8 +174,8 @@ class AutoBatchingMixin(object):
     MAX_IDEAL_BATCH_DURATION = 2
 
     # Batching counters
-    _effective_batch_size = 1
-    _smoothed_batch_duration = 0.0
+    _effective_batch_size = DEFAULT_EFFECTIVE_BATCH_SIZE
```

Comment:
I would just keep class variables but call them differently e.g. `_initial_effective_batch_size`. Then you don't need module constants and in `reset_batch_size` you can do:

```py
self._batch_size = self._initial_batch_size
```
.